src/components/visualizations/chart/HighChartRenderer.tsx

```diff
@@ -0,0 +1,224 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+import cloneDeep = require('lodash/cloneDeep');
+import get = require('lodash/get');
+import set = require('lodash/set');
+import isEqual = require('lodash/isEqual');
+import noop = require('lodash/noop');
+import partial = require('lodash/partial');
+import * as cx from 'classnames';
+
+import { VisualizationTypes } from '../../../constants/visualizationTypes';
+import Chart from './Chart';
+import Legend from './legend/Legend';
+import { TOP, LEFT, BOTTOM, RIGHT } from './legend/PositionTypes';
+
+export interface IHighChartsRendererProps {
+    chartOptions: any;
+    hcOptions: any;
+    height: number;
+    width: number;
+    legend: any;
+    legendRenderer(legendProps: any): void;
+    chartRenderer(chartProps: any): void;
+    afterRender(): void;
+    onLegendReady(legendItems: any): void;
+}
+
+export function renderChart(props: any) {
+    return <Chart {...props} />;
+}
+
+export function renderLegend(props: any) {
+    return <Legend {...props} />;
+}
+
+export default class HighChartRenderer extends React.PureComponent<IHighChartsRendererProps, any> {
```

Comment:
First, shouldn't it be HighChartsRenderer instead of HighChartRenderer?

I suggest to rewrite this as:

```typescript
export interface IHighChartRendererState {
    legendItemsEnabled: boolean[];
}

export default class HightChartsRenderer extends React.PureComponent<IHighChartsRendererProps, IHighChartsRendererState> {